spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -460,9 +465,28 @@ public Object doWithRetry(RetryContext context) throws RuntimeException {
 		final FixedSubscriberChannel bridge = new FixedSubscriberChannel(rh);
 		bridge.setBeanName("bridge." + name);
 
-		final KafkaMessageListenerContainer messageListenerContainer =
-				createMessageListenerContainer(properties, group, null, listenedPartitions, referencePoint);
+		Assert.isTrue(StringUtils.hasText(null) ^ !CollectionUtils.isEmpty(listenedPartitions),
```

Comment:
Good point. This is an artifact of inlining the old code. Initially we had support for passing both a topic or a list of partitions (for the never used `spyOn`),  so now it's always null - will fix.